test/test_im_onix.rb

```diff
@@ -458,12 +461,12 @@ def test_products_discount
       @product=@message.products.last
     end
 
-    should "have a supply with a price in Franec" do
-      supply = @product.supplies.first
+    should "have a supply with a price in France" do
+      frsupply = @product.supplies.select {|s| s[:territory].size == 1 and s[:territory].first == "FR"}
```

Comment:
Shouldn't the check on `s[:territory].size` be an assertion on the selected result rather than part of the select filter?

Reply:
Yes, you're right